keps/sig-node/20190226-pod-overhead.md

```diff
@@ -0,0 +1,310 @@
+pod cgroup.
+
+Users are not expected to manually set the pod resources; if a runtimeClass is being utilized,
+the manual value will be discarded. See RuntimeController for the proposal for setting these
+resources.
+
+Being able to specify resource requirements for a workload at the pod level instead of container level
+has been discussed, but isn't proposed in this KEP.
+
+In the event that pod-level requirements are introduced, pod overhead should be kept separate. This simplifies
+several scenarios:
+ - overhead, once added to the spec, stays with the workload, even if runtimeClass is redefined or removed.
+ - the pod spec can be referenced directly from scheduler, resourceQuote controller and kubelet, instead of referencing
+ a runtimeClass object which could have possibly been removed.
+
+
+### RuntimeClass changes
+
+Expand the runtimeClass type to include sandbox overhead, `Overhead *ResourceRequirements.`
+
+### RuntimeClass admission controller
+
+The pod resource overhead must be defined prior to scheduling, and we shouldn't make the user
+do it. To that end, we propose a new mutating admission controller: RuntimeClass. This admission controller
+is also proposed for the [native RuntimeClass scheduling KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/runtime-class-scheduling.md).
+
+In the scope of this KEP, The RuntimeClass controller will have a single job: set the pod overhead field in the
+workload's PodSpec according to the runtimeClass specified.
+
+It is expected that only the RuntimeClass controller will set Pod.Spec.Overhead. If a value is provided
+which does not match what is defined in the runtimeClass, the pod will be rejected.
+
+Going forward, I foresee additional controller scope around runtimeClass:
+ -validating the runtimeClass selection: This would require applying some kind of pod-characteristic labels
+ (runtimeClass selectors?) which would then be consumed by an admission controller and checked against known
+ capabilities on a per runtimeClass basis. This is is beyond the scope of this proposal.
+ -Automatic runtimeClass selection: A controller could exist which would attempt to automatically select the
+ most appropriate runtimeClass for the given pod. This, again, is beyond the scope of this proposal.
+
+### User Stories [optional]
+
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of the system.
+The goal here is to make this feel real for users without getting bogged down.
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they releate.
+
+### Risks and Mitigations
```

Comment:
@derekwaynecarr already touched on this, but IMO this should be treated as a new feature, i.e. with an indepedent feature gate from RuntimeClass. All the components that touch overhead should just ignore the overhead if none is set.